packages/types/src/tx/schema/transfer.ts

```diff
@@ -17,7 +17,16 @@ export class TransparentTransferMsgValue {
   @field(BigNumberSerializer)
   amount!: BigNumber;
 
-  constructor(data: TransparentTransferProps) {
+  constructor(data: TransferDataProps) {
+    Object.assign(this, data);
+  }
+}
+
+export class TransferMsgValue {
+  @field({ type: vec(TransferDataMsgValue) })
+  data!: TransferDataMsgValue[];
```

Comment:
Another question :smile: What does it do to pass an array of multiple values here, and how is that different from a batch tx?

Reply:
the `Transfer` Tx now contains an array of props, like `{ source, target, amount, token }`, so technically you could have a batch of transfers that each have multiple transfers within them. I'm just going according to the changes to the Tx, I have no idea why this would be needed to be honest!